packages/core/src/routes/interaction/index.ts

```diff
@@ -340,68 +335,29 @@ export default function interactionRoutes<T extends AnonymousRouter>(
 
       const profileVerifiedInteraction = await verifyProfile(tenant, accountVerifiedInteraction);
 
-      const interaction = isForgotPasswordInteractionResult(profileVerifiedInteraction)
+      const mandatoryProfileVerifiedInteraction = isForgotPasswordInteractionResult(
```

Comment:
if isForgotPasswordInterationResult return early?

Reply:
Could add me a todo here?  Maybe making all these verification steps a middleware could save lots of trouble. This looks silly.